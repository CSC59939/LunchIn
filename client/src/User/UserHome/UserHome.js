import React, { Component } from "react";
import "./UserHome.css";
import { firebase } from "../../Config";
import { Typography, Divider, Button } from "@material-ui/core";
import Logout from "@material-ui/icons/ExitToApp";
import CssBaseline from "@material-ui/core/CssBaseline";
import UserInputFoodChoices from "./UserInputFoodChoices/UserInputFoodChoices.js";
import Profile from "@material-ui/icons/Person";

class UserHome extends Component {
  constructor(props) {
    super(props);
    this.state = {
      user: {
        userUID: "",
        userName: ""
      }
    };
    this.firebaseListener = null;
  }
  componentDidMount() {
    this.authListener();
  }

  authListener() {
    this.fireBaseListener = firebase.auth().onAuthStateChanged(user => {
      if (user) {
        this.setState({
          user: {
            userUID: user.uid
          }
        });
      }
      firebase
        .database()
        .ref(`Users/${user.uid}/name`)
        .once(
          "value",
          snapshot => {
            if (snapshot.val() != null) {
              this.setState({
                userName: snapshot.val()
              });
            }
          },
          error => {
            console.log("Error: " + error.code);
          }
        );
    });
  }

  componentWillUnmount() {
    this.fireBaseListener && this.fireBaseListener();
    this.authListener = undefined;
  }

  render() {
    return (
      <div style={{ padding: "50px 200px" }}>
        <CssBaseline />
        {/*=============WELCOME USER HEADER=============*/}
        <div className="user-header">
          <Typography component="h2" variant="h3" style={{ flex: 1 }}>
            Welcome, {this.state.userName}
            <Button
              style={{ float: "right" }}
              onClick={() => {
                firebase.auth().signOut();
              }}
            >
              <Logout style={{ marginRight: "5px" }} /> Signout
            </Button>
            <Button
              style={{ float: "right" }}
              onClick={() => {
                this.props.history.push(`/user/profile`);
              }}
            >
              <Profile /> Profile
            </Button>
          </Typography>
        </div>
        <Divider />
        <UserInputFoodChoices userUID={this.state.user.userUID} />
      </div>
    );
  }
}

export default UserHome;
