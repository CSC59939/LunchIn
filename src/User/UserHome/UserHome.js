import React, { Component } from "react";
import "./UserHome.css";
import { firebase } from "../../Config";
import { Typography, Divider, Button } from "@material-ui/core";
import Logout from "@material-ui/icons/ExitToApp";
import UserInputFoodChoices from './UserInputFoodChoices/UserInputFoodChoices.js';
import Profile from "@material-ui/icons/Person";

class UserHome extends Component {
  constructor(props) {
    super(props);
    this.state = {
      user: {
        userUID: "",
        userName: ""
      }
    };
    this.firebaseListener = null;
  }
  componentDidMount() {
    this.authListener();
  }

  authListener() {
    this.fireBaseListener = firebase.auth().onAuthStateChanged(user => {
      if (user) {
        this.setState({
          user: {
            userName: user.displayName,
            userUID: user.uid
          }
        });
      }
    });
  }

  componentWillUnmount() {
    this.fireBaseListener && this.fireBaseListener();
    this.authListener = undefined;
  }
  render() {
    return (
      <div style={{ padding: "50px 200px" }}>
        {/*=============WELCOME USER HEADER=============*/}
        <div className="user-header">
          <Typography variant="display2" style={{ flex: 1 }}>
            Welcome, {this.state.user.userName}
            <Button
              style={{ float: "right" }}
              onClick={() => {
                firebase.auth().signOut();
              }}
            >
              <Logout style={{ marginRight: "5px" }} /> Signout
            </Button>

            <Button 
            style={{ float: "right" }}
            onClick={() => {
              this.props.history.push(`/user/profile`);
            }}>
                <Profile /> Profile  
            </Button>

          </Typography>
        </div>
        <Divider />
        <UserInputFoodChoices userUID={this.state.user.userUID}/>
      </div>
    );
  }
}

export default UserHome;
