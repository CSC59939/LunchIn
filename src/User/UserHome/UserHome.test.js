import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import UserHome from "./UserHome";

describe("UserHome Component", () => {
  describe("When provided with no props", () => {
    it("Should render correctly", () => {
      const reviewInstance = shallow(
        <UserHome />
      );
      expect(reviewInstance).toMatchSnapshot();
    });
  });
});
