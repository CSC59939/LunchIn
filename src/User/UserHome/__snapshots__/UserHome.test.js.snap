// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserHome Component When provided with no props Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserHome />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="user-header"
        >
          <WithStyles(Typography)
            style={
              Object {
                "flex": 1,
              }
            }
            variant="display2"
          >
            Welcome, 
            
            <WithStyles(Button)
              onClick={[Function]}
              style={
                Object {
                  "float": "right",
                }
              }
            >
              <pure(ExitToAppIcon)
                style={
                  Object {
                    "marginRight": "5px",
                  }
                }
              />
               Signout
            </WithStyles(Button)>
            <WithStyles(Button)
              onClick={[Function]}
              style={
                Object {
                  "float": "right",
                }
              }
            >
              <pure(PersonIcon) />
               Profile
            </WithStyles(Button)>
          </WithStyles(Typography)>
        </div>,
        <WithStyles(Divider) />,
        <UserInputFoodChoices
          userUID=""
        />,
      ],
      "style": Object {
        "padding": "50px 200px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <WithStyles(Typography)
            style={
              Object {
                "flex": 1,
              }
            }
            variant="display2"
          >
            Welcome, 
            
            <WithStyles(Button)
              onClick={[Function]}
              style={
                Object {
                  "float": "right",
                }
              }
            >
              <pure(ExitToAppIcon)
                style={
                  Object {
                    "marginRight": "5px",
                  }
                }
              />
               Signout
            </WithStyles(Button)>
            <WithStyles(Button)
              onClick={[Function]}
              style={
                Object {
                  "float": "right",
                }
              }
            >
              <pure(PersonIcon) />
               Profile
            </WithStyles(Button)>
          </WithStyles(Typography)>,
          "className": "user-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Welcome, ",
              "",
              <WithStyles(Button)
                onClick={[Function]}
                style={
                  Object {
                    "float": "right",
                  }
                }
              >
                <pure(ExitToAppIcon)
                  style={
                    Object {
                      "marginRight": "5px",
                    }
                  }
                />
                 Signout
              </WithStyles(Button)>,
              <WithStyles(Button)
                onClick={[Function]}
                style={
                  Object {
                    "float": "right",
                  }
                }
              >
                <pure(PersonIcon) />
                 Profile
              </WithStyles(Button)>,
            ],
            "style": Object {
              "flex": 1,
            },
            "variant": "display2",
          },
          "ref": null,
          "rendered": Array [
            "Welcome, ",
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <pure(ExitToAppIcon)
                    style={
                      Object {
                        "marginRight": "5px",
                      }
                    }
                  />,
                  " Signout",
                ],
                "onClick": [Function],
                "style": Object {
                  "float": "right",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "style": Object {
                      "marginRight": "5px",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                " Signout",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <pure(PersonIcon) />,
                  " Profile",
                ],
                "onClick": [Function],
                "style": Object {
                  "float": "right",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                " Profile",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "userUID": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="user-header"
          >
            <WithStyles(Typography)
              style={
                Object {
                  "flex": 1,
                }
              }
              variant="display2"
            >
              Welcome, 
              
              <WithStyles(Button)
                onClick={[Function]}
                style={
                  Object {
                    "float": "right",
                  }
                }
              >
                <pure(ExitToAppIcon)
                  style={
                    Object {
                      "marginRight": "5px",
                    }
                  }
                />
                 Signout
              </WithStyles(Button)>
              <WithStyles(Button)
                onClick={[Function]}
                style={
                  Object {
                    "float": "right",
                  }
                }
              >
                <pure(PersonIcon) />
                 Profile
              </WithStyles(Button)>
            </WithStyles(Typography)>
          </div>,
          <WithStyles(Divider) />,
          <UserInputFoodChoices
            userUID=""
          />,
        ],
        "style": Object {
          "padding": "50px 200px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <WithStyles(Typography)
              style={
                Object {
                  "flex": 1,
                }
              }
              variant="display2"
            >
              Welcome, 
              
              <WithStyles(Button)
                onClick={[Function]}
                style={
                  Object {
                    "float": "right",
                  }
                }
              >
                <pure(ExitToAppIcon)
                  style={
                    Object {
                      "marginRight": "5px",
                    }
                  }
                />
                 Signout
              </WithStyles(Button)>
              <WithStyles(Button)
                onClick={[Function]}
                style={
                  Object {
                    "float": "right",
                  }
                }
              >
                <pure(PersonIcon) />
                 Profile
              </WithStyles(Button)>
            </WithStyles(Typography)>,
            "className": "user-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Welcome, ",
                "",
                <WithStyles(Button)
                  onClick={[Function]}
                  style={
                    Object {
                      "float": "right",
                    }
                  }
                >
                  <pure(ExitToAppIcon)
                    style={
                      Object {
                        "marginRight": "5px",
                      }
                    }
                  />
                   Signout
                </WithStyles(Button)>,
                <WithStyles(Button)
                  onClick={[Function]}
                  style={
                    Object {
                      "float": "right",
                    }
                  }
                >
                  <pure(PersonIcon) />
                   Profile
                </WithStyles(Button)>,
              ],
              "style": Object {
                "flex": 1,
              },
              "variant": "display2",
            },
            "ref": null,
            "rendered": Array [
              "Welcome, ",
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <pure(ExitToAppIcon)
                      style={
                        Object {
                          "marginRight": "5px",
                        }
                      }
                    />,
                    " Signout",
                  ],
                  "onClick": [Function],
                  "style": Object {
                    "float": "right",
                  },
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "style": Object {
                        "marginRight": "5px",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  " Signout",
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <pure(PersonIcon) />,
                    " Profile",
                  ],
                  "onClick": [Function],
                  "style": Object {
                    "float": "right",
                  },
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  " Profile",
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "userUID": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
